zapcli/zap_helper.py

```diff
@@ -394,5 +394,5 @@ def html_report(self, file_path):
     @staticmethod
     def _write_report(report, file_path):
         """Write report to the given file path."""
-        with open(file_path, 'w') as f:
+        with open(file_path, mode='wb') as f:
             f.write(report)
```

Comment:
i.e. we could do:
```python
import six
...snip...

with open(file_path, mode='wb') as f:
    if six.PY3:
        report = report.encode('utf-8')
    f.write(report)
```

It's a bit ugly, but basically the string in Python 2.7 is a byte string, and the default system encoding is ascii, which is why it breaks with just mode `w` as it tries to encode a `utf-8` string ascii. Changing the mode to `wb` fixes this as then it doesn't try to encode it further when writing. But, in Python 3, the string is unicode so needs to be encoded as bytes before being written with mode `b` (but would work fine if the mode was just `w`).